***Solving a linear system Ax = b with (random) 7000 × 7000 matrix***

In [34]:
using LinearAlgebra
using Random

n = 7000
Random.seed!(123) 
A = rand(n, n)
b = rand(n)

function solve_linear_system()
    return A \ b
end

execution_time = @elapsed begin
    for _ in 1:70
        solve_linear_system()
    end
end

println("Julia Linear System Mean Time: ", execution_time / 70, " seconds")

Julia Linear System Solution Time: 5.590198401428571 seconds


first attempt: Julia Linear System Mean Time: 6.3472895128571425 seconds

second attempt: Julia Linear System Mean Time: 5.590198401428571 seconds

***Computing the eigenvalues/vectors of (random) 1200 × 1200 matrix A***

In [39]:
using LinearAlgebra
using Random

Random.seed!(123)
n = 1200

A = rand(n, n)

function compute_eigen()
    eigenvalues, eigenvectors = eigen(A)
end

execution_time = @elapsed begin
    for _ in 1:70
        compute_eigen()
    end
end

mean_time = execution_time / 70
println("Julia Eigenvalues and Eigenvectors Mean Time: $mean_time seconds")

Julia Eigenvalues and Eigenvectors Mean Time: 2.3041579842857143 seconds


first attempt: Julia Eigenvalues and Eigenvectors Mean Time: 2.6873683628571428 seconds

second attempt: Julia Eigenvalues and Eigenvectors Mean Time: 2.3041579842857143 seconds

***Computing the singular values/vectors of a (random) 2100 × 2100
matrix A***

In [57]:
using LinearAlgebra
using BenchmarkTools

function compute_svd(A)
    return svd(A)
end

n = 2100

A = randn(n, n)

total_time = 0.0
num_iterations = 70

for _ in 1:num_iterations
    total_time += @elapsed compute_svd(A)
end

mean_time = total_time / num_iterations

println("Mean time for Singular Value Decomposition: $mean_time seconds")

Mean time for 70 iterations: 4.75991864285714 seconds


first attempt: Mean time Singular Value Decomposition: 4.396511371428573 seconds

second attempt: Mean time for Singular Value Decomposition: 4.75991864285714 seconds